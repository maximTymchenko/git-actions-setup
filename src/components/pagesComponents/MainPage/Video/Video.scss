@import '../../../../styles/vars';

.video {
  position: relative;
  margin-top: 120px;
  border: solid 2px $secondary__dark;
  // background-image: url('../../../../assets/img/spectar-worker.jpg');
  // background-size: cover;
  // background-repeat: no-repeat;
  // background-color: $primary__light--80;
  @include section__size;
  height: 500px;
  display: flex;
  align-items: flex-end;

  @media screen and (max-width: $desktop) {
    margin-bottom: 50px;
  }

  &__background {
    position: absolute;
    top: 0;
    right: 0;
    left: 0;
    bottom: 0;
    width: 100%;
    // max-width: $primary__container;
    height: 100%;
    overflow: hidden;
  }

  &__iframe {
    position: absolute;
  }

  &::before {
    content: '';
    position: absolute;
    z-index: -1;
    top: 10px;
    left: 10px;
    width: 100%;
    height: 100%;
    border: 2px solid $secondary;
  }

  &::after {
    content: '';
    position: absolute;
    background-color: $primary__light--80;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
  }

  &__wrapper {
    position: relative;
    z-index: 8;
    margin-left: 30px;
    margin-bottom: 50px;

    @media screen and (max-width: $tablet) {
      margin-top: 120px;
    }

    @media screen and (max-width: $mobile) {
      margin-top: 40px;
      margin-left: 15px;
      margin-bottom: 25px;
    }
  }

  &__logo {
    margin-bottom: 15px;

    @media screen and (max-width: $mobile) {
      width: 90px;
    }
  }

  &__slogan {
    margin-bottom: 20px;
    max-width: 630px;
    font-family: 'OktaNeue-Medium';
    font-size: 22px;
    font-weight: 500;
    font-style: italic;
    color: $text;

    @media screen and (max-width: $mobile) {
      font-size: 16px;
    }
  }

  &__name {
    margin-bottom: 30px;
    display: inline-block;
    font-family: 'OktaNeue-Heavy';
    font-size: 15px;
    font-weight: 900;
    color: $text;

    @media screen and (max-width: $mobile) {
      margin-bottom: 10px;
      font-size: 15px;
    }
  }
}
