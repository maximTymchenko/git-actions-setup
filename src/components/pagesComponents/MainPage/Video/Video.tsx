import React, { FC } from 'react'
import Button from '../../../common/Button'
import './Video.scss'
import spectarBlack from '../../../../assets/img/spectar-black.svg'
import triangleRight from '../../../../assets/img/triangle-right.svg'
import video from '../../../../assets/video/video.mp4'

const Video: FC = () => {
  return (
    <div data-testid="video" className="video">
      <div className="video__intro">
        <video src={video} autoPlay muted loop />
      </div>
      <div className="video__wrapper">
        <img src={spectarBlack} alt="Spectar" className="video__logo" />
        <p className="video__slogan">
          “I could probably go into sales for you. It's incredible. Thanks for
          the great service.”
        </p>
        <span className="video__name">Anthony, CTO of Spectar.io</span>
        <Button
          icon={triangleRight}
          type={'primary'}
          className="button"
          desc={'Watch video'}
        >
          Watch video
        </Button>
      </div>
    </div>
  )
}

export default Video
