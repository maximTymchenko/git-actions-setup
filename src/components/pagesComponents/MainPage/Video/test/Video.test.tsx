import React from 'react';
import Video from '../Video';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

it('renders elem correctly', () => {
    const { getByTestId } = render(<Video></Video>);
    expect(getByTestId('video')).toBeTruthy();
});

it('renders without unexpected changes', () => {
    const tree = renderer.create(<Video></Video>).toJSON();
    expect(tree).toMatchSnapshot();
});
