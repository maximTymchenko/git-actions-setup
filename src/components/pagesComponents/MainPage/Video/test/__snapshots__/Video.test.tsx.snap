// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="video"
  data-testid="video"
>
  <div
    className="video__wrapper"
  >
    <img
      alt="Spectar"
      className="video__logo"
      src="test-file-stub"
    />
    <p
      className="video__slogan"
    >
      “I could probably go into sales for you. It's incredible. Thanks for the great service.”
    </p>
    <span
      className="video__name"
    >
      Anthony, CTO of Spectar.io
    </span>
    <button
      className="button button--primary button--undefined button--undefined"
      data-testid="button"
    >
      Watch video
      <img
        alt="Watch video"
        className="button__icon"
        data-testid="button__icon"
        src="test-file-stub"
      />
    </button>
  </div>
</div>
`;
